user_agent.go

```diff
@@ -3,41 +3,48 @@ package awsbase
 import (
 	"context"
 	"fmt"
-	"strings"
 
 	"github.com/aws/smithy-go/middleware"
 	smithyhttp "github.com/aws/smithy-go/transport/http"
 )
 
-func customUserAgentMiddleware(c *Config) middleware.BuildMiddleware {
-	return middleware.BuildMiddlewareFunc("CustomUserAgent",
+// Builds the user-agent string for APN
+func (apn APNInfo) BuildUserAgentString() string {
```

Comment:
I'm guessing no publicly available documentation for this format?

Reply:
It's under NDA